.github/workflows/ci.yaml

```diff
@@ -48,12 +50,17 @@ jobs:
         run: cargo build --verbose
       - name: Cargo Check
         run: cargo check
+      - name: Test Setup
+        run: cargo run --example test-setup
       - name: Unit Tests
         run: cargo test --lib
       - name: Doc Tests
         run: cargo test --doc
       - name: Integration Tests
         run: cargo test --tests
+      - name: Test Teardown
+        if: always()
```

Comment:
It will always run even if the job fails or is cancelled. I wanted to make sure that we wouldn't leak caches in those cases. I haven't tested it properly since the tests have always passed, however.